virtio-queue/src/queue/verification.rs

```diff
@@ -269,3 +502,57 @@ fn verify_spec_2_7_7_2() {
         // So we do not care
     }
 }
+
+/// Reads the idx field from the used ring in guest memory, as stored by Queue::add_used.
+/// Returns the value as u16.
+fn get_used_idx(
+    queue: &Queue,
+    mem: &SingleRegionGuestMemory,
+) -> Result<u16, vm_memory::GuestMemoryError> {
+    let addr =
+        queue
+            .used_ring
+            .checked_add(2)
+            .ok_or(vm_memory::GuestMemoryError::InvalidGuestAddress(
+                queue.used_ring,
+            ))?;
+    let val = mem.load(addr, Ordering::Acquire)?;
+    Ok(u16::from_le(val))
+}
+
+/// Kani proof harness for verifying the behavior of the `add_used` method of the `Queue`.
+///
+/// # Specification (VirtIO 1.3, Section 2.8.22: "Receiving Used Buffers From The Device")
```

Comment:
I think section 2.8 is for packed virtqueues whereas here we are only dealing with split virtqueues. I think the section should be 2.7.14 (see https://docs.oasis-open.org/virtio/virtio/v1.3/csd01/virtio-v1.3-csd01.html#x1-7100014)

Reply:
I think what we try to enforce here is `2.7.8.2 Device Requirements: The Virtqueue Used Ring`. I think we may need a comment like in firecraker in which we state all the things we do not verify, e.g., the order in which the used-ring indexes are updated.

Reply:
Not relevant here since proof moved to another PR

Reply:
We are **not** enforcing any of the following in the current proof therefore I don't think `2.7.8.2` applies:
1. The device MUST set len prior to updating the used idx.
2. The device MUST write at least len bytes to descriptor, beginning at the first device-writable buffer, prior to updating the used idx.
3. The device MAY write more than len bytes to descriptor. Note: There are potential error cases where a device might not know what parts of the buffers have been written. This is why len is permitted to be an underestimate: that’s preferable to the driver believing that uninitialized memory has been overwritten when it has not.

Reply:
added what we do not verify in doc